src/collector/event_collector.rs

```diff
@@ -0,0 +1,443 @@
+use std::{
+    boxed::Box,
+    future::Future,
+    pin::Pin,
+    sync::Arc,
+    task::{Context as FutContext, Poll},
+    time::Duration,
+};
+
+use futures::{future::BoxFuture, stream::Stream};
+use tokio::sync::mpsc::{
+    unbounded_channel,
+    UnboundedReceiver as Receiver,
+    UnboundedSender as Sender,
+};
+#[cfg(all(feature = "tokio_compat", not(feature = "tokio")))]
```

Comment:
```suggestion
#[cfg(not(feature = "tokio"))]
```

Reply:
Can you clarify why you're suggesting this? What I have currently matches the #[cfg] directive in all the existing collectors. Either way, having `tokio` disabled without `tokio_compat` enabled will cause a build failure, either because there's no imported `sleep` and `Sleep` (what I have now) or because this is trying to import something that doesn't exist (your suggestion).

Reply:
It is to be consistent with the other cfg restriction below. But such an issue should not he possible and not allowed in the `build.rs` file no?

Reply:
build.rs only checks for a missing TLS backend, it doesn't check that you have one of either `tokio` or `tokio_compat` enabled.

I'm fine with changing this but I'd rather do it in a separate PR and change all the collectors at the same time if so, and leave this the same as the others for now. How does that sound?